File: trees/inOrder/inOrder.ts
// Importamos el tipo Arbol desde su archivo correspondiente
import { Arbol } from "../type"

// Definimos una función llamada "inOrder" que toma un nodo raíz (de tipo Arbol o nulo) como argumento
export const inOrder = (root: Arbol | null): void => {
  // Si el nodo raíz no es nulo, realizamos el recorrido en orden (in-order)
  if (root !== null) {
    // Llamamos recursivamente a "inOrder" para el subárbol izquierdo
    inOrder(root.izquierdo)
    // Imprimimos el valor del nodo actual
    console.log(root.valor)
    // Luego, llamamos recursivamente a "inOrder" para el subárbol derecho
    inOrder(root.derecho)
  }
}

File: trees/inOrder/inOrder.pas
// Definimos un procedimiento llamado "inOrder" que toma un puntero a un nodo de árbol (de tipo "arbol") como argumento
procedure inOrder(nodo: arbol);
begin
    // Si el nodo no es nulo (es decir, existe), realizamos el recorrido en orden (in-order)
    if (nodo <> nil) then
    begin
        // Llamamos recursivamente a "inOrder" para el subárbol izquierdo
        inOrder(nodo^.hi);
        // Imprimimos el valor del nodo actual
        writeln(nodo^.dato);
        // Luego, llamamos recursivamente a "inOrder" para el subárbol derecho
        inOrder(nodo^.hd)
    end
end;
